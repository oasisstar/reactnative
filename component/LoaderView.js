import React, { Component } from 'react';
import	{
            StyleSheet,
            View,
            Text,
            ActivityIndicator
        } from 'react-native';

class LoaderView extends Component {
    render() {
        return(
            <View style={styles.container}>
                <ActivityIndicator/> 
                <Text style={styles.loadText}>Loading data...</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center',
    },
    
    loadText: {
        fontSize: 20
    }
});

export default LoaderView;