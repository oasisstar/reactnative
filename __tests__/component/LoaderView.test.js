// __tests__/LoaderView-test.js
import React from 'react';
import LoaderView from '../../component/LoaderView';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<LoaderView />).toJSON();
  expect(tree).toMatchSnapshot();
});