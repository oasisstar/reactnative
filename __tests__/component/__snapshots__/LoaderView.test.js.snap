// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Loading data...
  </Text>
</View>
`;
